frontend/src/components/Header.tsx

```diff
@@ -1,20 +1,96 @@
-import { NavLink } from "react-router-dom";
-import { NAV } from "../utils/consts";
+import { useState } from "react";
+import { NavLink, useNavigate } from "react-router-dom";
+import { AiOutlineClose, AiOutlineMenu } from "react-icons/ai";
+
 import Login from "./Login";
+import { NAV, NAV_ITEMS } from "../lib/consts";
+import { Button } from "./UI/button";
+import { useAppDispatch } from "../hooks/redux";
+import { toggleNav } from "../store/uiSlice";
 
 function Header() {
+  const [isNav, setIsNav] = useState(false);
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+
+  const handleNav = () => {
+    setIsNav(prev => !prev);
+    dispatch(toggleNav());
+  };
+
+  const handleNavigateHome = () => {
```

Comment:
I think there is no need in this function. You can use button inside NavLink tag and get rid of it. Smth like this <NavLink to="/" aria-label="Home">
  <Button
    variant
    className
  />
</NavLink>

Reply:
<NavLink to="/" aria-label="Home">
  <Button
    variant
    className
  />
</NavLink>

Reply:
"<NavLink to="/" aria-label="Home">
  <Button
    variant
    className
  />
</NavLink>"

Reply:
NavLink to="/" aria-label="Home"
  Button
    variant
    className
  
/NavLink